meshtastic/interfaces/ble/notifications.py

```
@@ -228,3 +257,528 @@ def _get_callback(self, characteristic: str) -> Callable[[Any, Any], None] | Non
+                    "Unable to start FROMNUM notifications for %s: %s; falling back to polling reads.",
+                    FROMNUM_UUID,
+                    err,
+                )
+                return
+            else:
+                if self._notification_manager._get_callback(FROMNUM_UUID) is None:
+                    self._notification_manager._subscribe(FROMNUM_UUID, ingress_handler)
+                self.fromnum_notify_enabled = True
+                return
+
+    def log_radio_handler(self, _: Any, b: bytes | bytearray) -> str | None:
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_`

(ANN401)